### DISCUSSION TITLE
Server Actions: Why are there no timeout errors in nextjs server actions

### DISCUSSION THREAD

**Discussion by maimunahmed2** [vercel/next.js#63590] [Help]
I am building a nextjs application that downloads a file into the server and saves it into firebase cloud storage. Initially I made an api that will do all this stuff which worked when in development but gave a timeout error when deployed to the vercel and ran in production.

But as I modified the code into a server action, there were no timeout errors both in deployment and production. I even tried downloading very big files but still there were no timeouts. Not sure why this is happening. Do you guys have any idea?

Edit: Fixed small typos ;)

**Comment by gaburielcasado** [comment]
Bump. It's pretty bad. I don't understand why Vercel does not show server action names.
All logging is pretty much worthless without that information...
It's such a downgrade from API routes.
